src/tools/ceph_objectstore_tool.cc

```diff
@@ -415,6 +413,22 @@ int mark_pg_for_removal(ObjectStore *fs, spg_t pgid, ObjectStore::Transaction *t
 #pragma GCC diagnostic pop
 #pragma GCC diagnostic warning "-Wpragmas"
 
+template<typename Func>
+void wait_until_done(ObjectStore::Transaction* txn, Func&& func)
```

Comment:
Why is this mechanism needed now?

Reply:
see the commit message of https://github.com/ceph/ceph/pull/20593/commits/42f310f253a658ec8c600a8406084ec91ecd49ba